src/test/ui/resolve/resolve-type-param-in-item-in-trait.stderr

```diff
@@ -23,7 +23,7 @@ error[E0401]: can't use generic parameters from outer function
   --> $DIR/resolve-type-param-in-item-in-trait.rs:23:28
    |
 LL | trait TraitC<A> {
-   |              - type variable from outer function
+   |              - type parameter from outer function
```

Comment:
The diagnostic makes no sense here as there's no outer function, just an outer trait.

Reply:
I believe there's already a ticket for this case.

Reply:
@estebank can you cc it perhaps?

Reply:
CC #37892, #56794, #53241

Reply:
Thanks; I skimmed those issues and they all seem a bit different. I think the fix for this one is to customize the diagnostic based on what the parent def_id refers to... e.g. say "function" if it is one, and "trait" if it is one...